1.1.16

ChrisBarker-NOAA opened the pull request:
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

@conda-forge-admin, please rerender

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ ``requirements: run: gridded >= 0.7.1`` should not contain a space between relational operator and the version, i.e. ``gridded >=0.7.1``

For **recipe/meta.yaml**:

 * ℹ️ top-level output has some malformed specs:
- In section run: `gridded >= 0.7.1`
Requirement spec fields should match the syntax `name [version [build]]`to avoid known issues in conda-build. For example, instead of `name =version=build`, use `name version.* build`. There should be no spaces between version operators and versions either: `python >= 3.8` should be `python >=3.8`.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/15007707334](https://github.com/conda-forge/conda-forge-webservices/actions/runs/15007707334). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

ChrisBarker-NOAA commented:
@conda-forge/help-python-c:

This is very, very odd -- it's all seemingly building fine, but then when the tests run -- it segfaults out. 

And in the same way on all platforms and python versions. -- What the Heck?

Building locally with the `build-locally.py` script -- I get the same thing -- segfaults on the tests.

But: -- if I go find the broken package that was created, and hand-install it into a hand-bud test environment, it does work just fine. So something about the test environment is different / broken. And I can't see what.

Any ideas? anyone ever seen this before?

NOTE: I first encountered this in December, and got stuck and sidetracked -- since then, I've updated to a new version of the package, updated dependencies, re-rendered, all of that -- with consistently the same segfaults ??!!??

NOTE 2: the import works fine -- the segfault occurs when running the actual unit tests. I'm doing that, because this is a big complicated package with a lot of Cython, Numpy dependency, etc, and in our first go around I accidentally published a package that was broken on Windows -- so it's a really good idea to actually run the full test suite (could be part of the test suite, but that wouldn't help the segfault problem ...)

ChrisBarker-NOAA commented:
New NOTE:

We found the built packages (one of them, anyway) in the artifacts from the build. If we download and install that (OS-X, anyway), and run it in a local hand-built environment, all is well -- so it seems the package is fine, but the test environment is fatally broken. On all platforms -- weird! 

And I found the test environment created by running teh build locally, and when I try to run the tests in there, I get the segfault -- but that same package works fine running in a local hand-built environment.

This is all very odd -- we can probably simply turn those tests off, but I'd really like to keep them ...

ChrisBarker-NOAA commented:
More info:

We've isolated the segfault to one Cython function (that calls a C++ function under the hood) -- so it may be a bug in our code.

However, that code hasn't changed in years, and it's worked without error all that time, and currently works in all our test and operational environments -- Windows, OS-X and Linux, and works when we build the conda package locally, and works when we run the conda package that is built here on the CI, in a hand-created environment.

So really hard for us to debug, if it is a subtle thing in our code.

Also -- it's segfaulting on all three platforms and multiple python versions -- so what's special about conda's test environment(s) that would trigger this???

JamesMakela-NOAA commented:
Among a few others, the Windows build is failing on this basic pytest: 

`unit_tests.test_cy.test_cy_random_mover.TestRandom.test_seed`

This should be pretty basic functionality, and it appears that the stdlib function `srand(1)` is not reinitializing the random seed to its previous value.  According to the cpp documentation:

*"If seed is set to 1, the generator is reinitialized to its initial value and produces the same values as before any call to rand or srand."*

Reference: https://cplusplus.com/reference/cstdlib/srand/

This is not happening in our local environments, so appears to be something specific to the Windows build environments.

ChrisBarker-NOAA commented:
@conda-forge-admin: please rerender

ChrisBarker-NOAA commented:
@conda-forge-admin: please rerender

ChrisBarker-NOAA commented:
closing in favor of #10

ChrisBarker-NOAA commented:
closed in favor of #10
